nsd_exporter.go

```diff
@@ -54,8 +56,40 @@ func (c *NSDCollector) Describe(ch chan<- *prometheus.Desc) {
 		ch <- metric.desc
 	}
 }
+func CollectZone(str string) string {
+	var re = regexp.MustCompile(`(?m)zone:\s+(?P<zone>.*)\n\s+state:\s(?P<state>\w+)\n\s+served-serial:\s(?P<served_serial>(\w+)|("(?P<served_soa>\d+)\ssince\s(?P<served_soa_date>.*)"))\n\s+commit-serial:\s(?P<commit_serial>(\w+)|("(?P<commit_soa>\d+)\ssince\s(?P<commit_soa_date>.*)"))\n\s+((wait:\s"(?P<wait>.*)")|notified-serial:\s"(?P<notified_serial>.*)"\n\s+transfer:\s"TCP\sconnected\sto\s(?P<transfer_ip>.*)")?`)
+	lmatch := re.FindAllStringSubmatch(str, -1)
+	var ret string
+	var zone string
+	var soa string
+	for _, match := range lmatch {
+		for i, name := range re.SubexpNames() {
+			if i != 0 && name == "zone" {
+				zone = match[i]
+			} else if name == "served_soa" {
+				if match[i] == "" {
+					soa = "0"
+				} else {
+					soa = match[i]
+				}
+			}
+		}
+		ret = ret + "zone.soa." + zone + "=" + soa + "\n"
```

Comment:
I see that you're converting the `zonestatus` output into something similar to `stats`. Not sure how I feel about this double conversion here. Are you just trying to leverage the existing metrics parsing code?

Reply:
Actually looking at this deeper, can you explain exactly what you're trying to measure here? It looks like `zonestatus` only points out whether the zone is in transfer or not.